agent/lib/kontena/workers/weave_worker.rb

```diff
@@ -71,8 +73,12 @@ def weave_attach(container)
 
     # @param [String] container_id
     # @param [String] overlay_cidr
-    def attach_overlay(container_id, overlay_cidr)
-      network_adapter.exec(['--local', 'attach', overlay_cidr, '--rewrite-hosts', container_id])
+    def attach_overlay(container)
+      if container.overlay_suffix != OVERLAY_SUFFIX
```

Comment:
This is a bit scary... we need to handle this better in the future when we a grid can have a custom subnet.